File: src/main/java/de/pbauerochse/worklogviewer/youtrack/v20174/types/FieldType.java
package de.pbauerochse.worklogviewer.youtrack.v20174.types;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class FieldType {

    private final String id;

    @JsonCreator
    public FieldType(@JsonProperty("id") String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    @JsonProperty("$type")
    public String getType() {
        return "jetbrains.charisma.persistence.customfields.FieldType";
    }
}

File: application/src/main/java/de/pbauerochse/worklogviewer/youtrack/v20174/types/FieldType.kt
package de.pbauerochse.worklogviewer.youtrack.v20174.types

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

@Deprecated("")
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
class FieldType @JsonCreator
constructor(@param:JsonProperty("id") val id: String) {

    val type: String
        @JsonProperty("\$type")
        get() = "jetbrains.charisma.persistence.customfields.FieldType"
}
